File: leetcode/3131-find-the-integer-added-to-array-i/addedInteger_test.go
package leetcode

import (
	"fmt"
	"path/filepath"
	"testing"

	"github.com/edram/leetcode/util"
)

type AddedIntegerCase struct {
	Input struct {
		Nums1 []int
		Nums2 []int
	}

	Output int
}

func TestAddedInteger(t *testing.T) {
	var casePath, _ = filepath.Abs("./cases.json")
	payload, err := util.LoadCases[[]AddedIntegerCase](casePath)
	if err != nil {
		panic(err)
	}

	for i, testCase := range *payload {
		fmt.Printf("正在执行第 %d 个测试用例\n", i+1)
		fmt.Printf("string: %v ==> %v \n", testCase.Input, testCase.Output)

		result := addedInteger(testCase.Input.Nums1, testCase.Input.Nums2)
		if result != testCase.Output {
			t.Errorf("期望 %v, 结果 %v", testCase.Output, result)
		}
	}
}

File: leetcode/3131-find-the-integer-added-to-array-i/addedInteger_test.rs
use std::{env::current_dir, path::Path};

use serde::Deserialize;

use crate::{util::load_cases, Solution};

#[derive(Deserialize, Debug)]
struct Input {
    nums1: Vec<i32>,
    nums2: Vec<i32>,
}

#[derive(Deserialize, Debug)]
struct Case {
    input: Input,
    output: i32,
}

#[test]
fn test_find_the_integer_added_to_array_i() {
    let current_path = current_dir().unwrap().join(file!());
    let dir = current_path.parent().unwrap();

    let cases: Vec<Case> = load_cases(Path::new(&dir.join("./cases.json")));

    for case in &cases {
        assert_eq!(
            Solution::added_integer(case.input.nums1.clone(), case.input.nums2.clone()),
            case.output
        );
    }
}
